Bump pylint to 3.3.7, update changelog (#10367)
pylint-dev/pylint: doc/whatsnew/3/3.3/index.rst

.. towncrier release notes start

What's new in Pylint 3.3.7?
---------------------------
Release date: 2025-05-04

False Positives Fixed
---------------------

- Comparisons between two calls to `type()` won't raise an ``unidiomatic-typecheck`` warning anymore, consistent with the behavior applied only for ``==`` previously.

  Closes #10161 (`#10161 <https://github.com/pylint-dev/pylint/issues/10161>`_)

Other Bug Fixes
---------------

> - Fixed a crash when importing a class decorator that did not exist with the same name than a class attribute after the class definition.

Reviewer: ```suggestion
- Fixed a crash when importing a class decorator that did not exist with the same name as a class attribute after the class definition.
```

  Closes #10105 (`#10105 <https://github.com/pylint-dev/pylint/issues/10105>`_)

- Fix a crash caused by malformed format strings when using `.format` with keyword arguments.

  Closes #10282 (`#10282 <https://github.com/pylint-dev/pylint/issues/10282>`_)

> - Using a slice as a class decorator now raise a 'not-callable' message instead of crashing pylint. A lot of checks that dealt with decorators (too many to list) are now shortcut if the decorator can't immediately be inferred to a function or class definition.

Reviewer: ```suggestion
- Using a slice as a class decorator now raises a ``not-callable`` message instead of crashing. A lot of checks that dealt with decorators (too many to list) are now shortcut if the decorator can't immediately be inferred to a function or class definition.
```

  Closes #10334 (`#10334 <https://github.com/pylint-dev/pylint/issues/10334>`_)

Other Changes
-------------

> - The algorithm used for ``no-member`` suggestions is now more efficient and cut the

Reviewer: ```suggestion
- The algorithm used for ``no-member`` suggestions is now more efficient and cuts the
```
